catroid/src/main/java/org/catrobat/catroid/cloudmessaging/CloudMessaging.java

```diff
@@ -54,6 +69,50 @@ public void showNotification(CloudMessage cloudMessage) {
 		}
 	}
 
+	public static boolean initialize(Context context) {
+		if (isCloudNotificationsEnabled(context)) {
+			subscribeToFirebaseServices(context);
+			return true;
+		}
+		unsubscribeFromFirebaseServices(context);
+		return false;
+	}
+
+	@VisibleForTesting
+	public static void setIsCloudNotificationsEnabled(boolean isEnabled) {
+		isCloudNotificationsEnabled = isEnabled;
+	}
+
+	private static void subscribeToFirebaseServices(Context context) {
+		FirebaseMessaging.getInstance().subscribeToTopic(DEFAULT_TOPIC);
+		FirebaseMessaging.getInstance().subscribeToTopic(getLanguageCode(context));
+		FirebaseMessaging.getInstance().subscribeToTopic(getCountryCode(context));
+	}
+
+	private static void unsubscribeFromFirebaseServices(Context context) {
+		FirebaseMessaging.getInstance().unsubscribeFromTopic(DEFAULT_TOPIC);
+		FirebaseMessaging.getInstance().unsubscribeFromTopic(getLanguageCode(context));
+		FirebaseMessaging.getInstance().unsubscribeFromTopic(getCountryCode(context));
+	}
+
+	private static boolean isCloudNotificationsEnabled(Context context) {
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+		return sharedPreferences.getBoolean(SettingsActivity.SETTINGS_CLOUD_NOTIFICATIONS, false) && isCloudNotificationsEnabled;
+	}
+
+	private static String getLanguageCode(Context context) {
+		String languageCode = context.getResources().getConfiguration().locale.getCountry();
```

Comment:
> Do I have to create key-value pairs for strings.xml?

No, Crowdin would remove these strings anyway.
We can make sure that there is a language tag in every strings.xml file by using a unit test.

> Which of the 2 methods (or any other method) should I use? (for un/subscribing)

Both, I guess. With an additional change
- "whenever the user opens the ~app~ _MainMenuActivity_"

Changing the app language through the app (https://github.com/Catrobat/Catroid/pull/2332) will finish all activities and open the MainMenuActivity.

@cheidenreich, what do you think as well?